eth/handler_eth.go

```diff
@@ -42,11 +43,13 @@ func (n NilPool) Get(common.Hash) *types.Transaction              { return nil }
 func (n NilPool) GetRLP(common.Hash) []byte                       { return nil }
 func (n NilPool) GetMetadata(hash common.Hash) *txpool.TxMetadata { return nil }
 
-func (h *ethHandler) TxPool() eth.TxPool {
-	if h.noTxGossip {
-		return &NilPool{}
+func (h *ethHandler) TxPool(peer *p2p.Peer) (eth.TxPool, bool) {
+	if h.noTxGossip ||
+		(h.txGossipTrustedPeersOnly && !peer.Trusted()) ||
+		(h.txGossipNetRestrict != nil && !h.txGossipNetRestrict.ContainsAddr(peer.Node().IPAddr())) {
+		return &NilPool{}, false
 	}
-	return h.txpool
+	return h.txpool, true
 }
```

Comment:
I think if you add method
```go
func (h *ethHandler) txGossipAllowed(peer *p2p.Peer) bool {
	return h.noTxGossip ||
		(h.txGossipTrustedPeersOnly && !peer.Trusted()) ||
		(h.txGossipNetRestrict != nil && !h.txGossipNetRestrict.ContainsAddr(peer.Node().IPAddr()))
}
```
then you don't need to return a `bool` here and below you can use this method and remove the 
additional field `allowedForTxGossip bool` of the `Peer`.

```suggestion
func (h *ethHandler) TxPool(peer *p2p.Peer) eth.TxPool {
	if h.txGossipAllowed(peer) {
		return new(NilPool)
	}
	return h.txpool
}
```

Reply:
Good idea, I added it.